src/Component.js

```diff
@@ -48,19 +50,30 @@ export class DebounceInput extends React.PureComponent {
 
 
   componentWillMount() {
-    this.createNotifier(this.props.debounceTimeout);
+    const {
+      debounceTimeout,
+      debounceOptions
+    } = this.props;
+    this.createNotifier(debounceTimeout, debounceOptions);
   }
 
 
-  componentWillReceiveProps({value, debounceTimeout}) {
+  componentWillReceiveProps({value, debounceTimeout, debounceOptions}) {
     if (this.isDebouncing) {
       return;
     }
     if (typeof value !== 'undefined' && this.state.value !== value) {
       this.setState({value});
     }
-    if (debounceTimeout !== this.props.debounceTimeout) {
-      this.createNotifier(debounceTimeout);
+    const {
+      debounceTimeout: debounceTimeoutCurrent,
+      debounceOptions: debounceOptionsCurrent
+    } = this.props;
+    if (
+      debounceTimeout !== debounceTimeoutCurrent ||
+      debounceOptions !== debounceOptionsCurrent
```

Comment:
comparing `{} !== {}` would trigger updates on each render. Unfortunately for objects we need to use shallow object compare instead.

Reply:
I am not sure when `{} !== {}` is actually going to happen. The default is `undefined`. If someone sets the property, the object will stay the same until they change it.